crates/trie/parallel/src/proof.rs

```diff
@@ -48,11 +51,12 @@ pub struct ParallelProof<Factory> {
     collect_branch_node_masks: bool,
     /// Thread pool for local tasks
     thread_pool: Arc<rayon::ThreadPool>,
+    metrics: ParallelTrieMetrics,
```

Comment:
should we feature-gate the field definition too?

Reply:
tbh not sure why there's a metrics feature here, setting metrics does not have much overhead, so would actually be in favor of removing the feature flag

Reply:
agreed, we use the `ParallelTrieTracker` either way, and reporting metrics is cheap

Reply:
the only reason can be that we won't bring `metrics` and `reth-metrics` deps if you don't have metrics enabled